conan/tools/apple/xcodebuild.py

```diff
@@ -26,18 +28,29 @@ def _sdkroot(self):
             sdk = "{}{}".format(self._sdk, self._sdk_version)
         return "SDKROOT={}".format(sdk) if sdk else ""
 
-    def build(self, xcodeproj, target=None):
+    def build(self, xcodeproj, target=None, configuration=None, build_options=[]):
         """
         Call to ``xcodebuild`` to build a Xcode project.
 
         :param xcodeproj: the *xcodeproj* file to build.
         :param target: the target to build, in case this argument is passed to the ``build()``
                        method it will add the ``-target`` argument to the build system call. If not passed, it
                        will build all the targets passing the ``-alltargets`` argument instead.
+        :param configuration: the configuration to build, defaults to profile's ``settings.build_type``.
+        :param build_options: arbitrary options (list of strings) to be passed to ``xcodebuild`` as is.
         :return: the return code for the launched ``xcodebuild`` command.
         """
         target = "-target {}".format(target) if target else "-alltargets"
-        cmd = "xcodebuild -project {} -configuration {} -arch {} " \
-              "{} {} {}".format(xcodeproj, self._build_type, self._arch, self._sdkroot,
-                                self._verbosity, target)
+        build_config = configuration or self._build_type
+        cmd = "xcodebuild -project '{}' -configuration {} -arch {} " \
+              "SYMROOT='{}' OBJROOT='{}' RUN_CLANG_STATIC_ANALYZER=NO " \
+              "{} {} {}".format(xcodeproj, build_config, self._arch,
+                                self._conanfile.build_folder, self._conanfile.build_folder,
+                                self._sdkroot, self._verbosity, target)
+
+        deployment_target_key = xcodebuild_deployment_target_key(self._os)
+        if deployment_target_key and self._os_version:
```

Comment:
This is probably the main and necessary fix?
I understand that if this is not provided, it will fail to build for iOS, watchOS, etc?

Reply:
if it's not provided, the setting from the Xcode project will be used, which may be not what the library user desires. Since profile defines OS version, it's natural to use it.